src/ynEIGEN/withdrawalsProcessor.sol

```diff
@@ -0,0 +1,364 @@
+        address _delegationManager
+    ) Ownable(_owner) {
+        if (
+            _withdrawalQueueManager == address(0) ||
+            _tokenStakingNodesManager == address(0) ||
+            _assetRegistry == address(0) ||
+            _ynStrategyManager == address(0) ||
+            _delegationManager == address(0)
+        ) revert InvalidInput();
+
+        withdrawalQueueManager = IWithdrawalQueueManager(_withdrawalQueueManager);
+        tokenStakingNodesManager = ITokenStakingNodesManager(_tokenStakingNodesManager);
+        assetRegistry = IAssetRegistry(_assetRegistry);
+        ynStrategyManager = IYieldNestStrategyManager(_ynStrategyManager);
+        delegationManager = IDelegationManager(_delegationManager);
+
+        minNodeShares = 1 ether;
+        minPendingWithdrawalRequestAmount = 0.1 ether;
+    }
+
+    //
+    // Processor functions - view
+    //
+
+    function shouldQueueWithdrawals() view returns (bool) {
+        return
+            withdrawalQueueManager.pendingRequestedRedemptionAmount()
+            - totalQueuedWithdrawals
+            - redemptionAssetsVault.availableRedemptionAssets()
+            > minPendingWithdrawalRequestAmount
+    }
+
+    // _minNodeShares -- the second highest shares balance (so that we only withdraw from the highest balance node)
+    /// @notice Gets the arguments for `queueWithdrawals`
+    /// @param _asset The asset to withdraw:
+    ///               the asset with the highest balance
+    /// @param _minNodeShares The second highest shares balance:
+    ///                       meaning that we only withdraw from the highest balance node(s)
+    /// @param _maxSharesToWithdraw The maximum amount of shares that can be withdrawn from a node:
+    ///                             we try to bring the highest balance node(s) to have the same balance
+    ///                             as the second highest balance node(s)
+    function getQueueWithdrawalsArgs() view returns (IERC20 _asset, uint256 _minNodeShares, uint256 _maxSharesToWithdraw) {
+
+        // get `_asset`
+        {
+            IERC20[] memory _assets = assetRegistry.getAssets();
+            uint256[] memory _balances = yneigen.assetBalances(_assets);
+
+            uint256 _highestBalance;
+            uint256 _assetsLength = _assets.length;
+            for (uint256 i = 0; i < _assetsLength; ++i) {
+                if (_balances[i] > _highestBalance) {
+                    _highestBalance = _balances[i];
+                    _asset = _assets[i];
+                }
+            }
+        }
+
+        ITokenStakingNode[] memory _nodes = tokenStakingNodesManager.getAllNodes();
+        uint256 _nodesLength = _nodes.length;
+        uint256[] memory _nodesShares = new uint256[](_nodesLength);
+
+        // get `_minNodeShares`
+        {
+            IStrategy _strategy = ynStrategyManager.strategies(_asset);
+
+            uint256 _highest;
+            uint256 _secondHighest;
```

Comment:
I do see one issue here:

The initial case is all have the same number of shares - that's wha it is on-chain now, for the sake of example 500 OETH on each node.

so the getQueueWithdrawalsArgs willl return that number - 500.

Now when it goes in to execute with _minNodeShares =  the OETH amount on all nodes, it won't be able to withdaw anything because they're all hitting the floor
so directionally it looks right but i see an issue with this edge case